test/common-test.js

```diff
@@ -57,10 +68,11 @@ module.exports = {
             port.name = name;
         });
 
-        return component;
+        return node ? node : component;
```

Comment:
This is a little tricky since the function can return a node if it exists or a component if it does not.  Can you add a  comment explaining what's going on with this?